pallets/liquid-staking-v2/src/lib.rs

```diff
@@ -319,13 +416,29 @@ mod pallet {
             era_index: EraIndex,
         ) -> DispatchResultWithPostInfo {
             T::BridgeOrigin::ensure_origin(origin)?;
-            let current_era_index = Self::current_era();
-            ensure!(current_era_index < era_index, Error::<T>::EraAlreadyPushed,);
+            let current_era = Self::current_era();
+            ensure!(current_era < era_index, Error::<T>::EraAlreadyPushed,);
+
+            let (op_type, status, amount) = Self::matching_pool(current_era).summary();
```

Comment:
if we move this logic after `CurrentEra::<T>::put(era_index);`, and use `Self::matching_pool(previous_era)`, does it effect the logic, can you explain the different?

Reply:
@MrPai it should be the same actually. If we put them in the same transactional call. But in previous version `trigger_new_era` only contain era update logic. I've put them together, not sure which one is the best because there are pros & cons